--- a/gui/greeter/greeter_geometry.c	2010-06-02 20:06:25.000000000 -0500
+++ b/gui/greeter/greeter_geometry.c	2011-08-14 19:27:24.571295838 -0500
@@ -594,10 +594,24 @@
  
   if (req->width > 0 && req->height > 0)
     {
+    if (item->width_type == GREETER_ITEM_SIZE_SCALE && item->height_type == GREETER_ITEM_SIZE_SCALE)
+      {
+      set_width = parent_width;
+      set_height = (req->height * set_width) / req->width;
+
+      if (set_height < parent_height)
+        {
+        set_height = parent_height;
+        set_width = (req->width * set_height) / req->height;
+        }
+      }
+    else
+      {
       if (item->width_type == GREETER_ITEM_SIZE_SCALE && set_height > 0)
         set_width = (req->width * set_height) / req->height;
       else if (item->height_type == GREETER_ITEM_SIZE_SCALE && set_width > 0)
         set_height = (req->height * set_width) / req->width;
+      }
     }
   
   if (set_width > 0)
