diff --git a/src/VBox/RDP/Makefile.kmk b/src/VBox/RDP/Makefile.kmk
new file mode 100644
index 0000000..76d4bb2
--- /dev/null
+++ b/src/VBox/RDP/Makefile.kmk
@@ -0,0 +1,28 @@
+# $Id: Makefile.kmk $
+## @file
+# Top-level makefile for src/VBox/RDP.
+#
+
+#
+# Copyright (C) 2006-2010 Oracle Corporation
+#
+# This file is part of VirtualBox Open Source Edition (OSE), as
+# available from http://www.virtualbox.org. This file is free software;
+# you can redistribute it and/or modify it under the terms of the GNU
+# General Public License (GPL) as published by the Free Software
+# Foundation, in version 2 as it comes in the "COPYING" file of the
+# VirtualBox OSE distribution. VirtualBox OSE is distributed in the
+# hope that it will be useful, but WITHOUT ANY WARRANTY of any kind.
+#
+
+SUB_DEPTH = ../../..
+include $(KBUILD_PATH)/subheader.kmk
+
+
+ifdef VBOX_WITH_VRDP_RDESKTOP
+ include $(PATH_SUB_CURRENT)/client/Makefile.kmk
+endif
+
+
+include $(KBUILD_PATH)/subfooter.kmk
+
diff --git a/src/libs/Makefile.kmk b/src/libs/Makefile.kmk
index 81e76d3..64b67ff 100644
--- a/src/libs/Makefile.kmk
+++ b/src/libs/Makefile.kmk
@@ -56,9 +56,9 @@ if !defined(VBOX_ONLY_SDK) \
 endif
 
 # libjpeg for VRDP video redirection
-if defined(VBOX_WITH_VRDP) && defined(VBOX_WITH_VRDP_VIDEO_CHANNEL)
- include $(PATH_SUB_CURRENT)/jpeg-8a/Makefile.kmk
-endif
+#if defined(VBOX_WITH_VRDP) && defined(VBOX_WITH_VRDP_VIDEO_CHANNEL)
+# include $(PATH_SUB_CURRENT)/jpeg-8a/Makefile.kmk
+#endif
 
 # Main related things - XPCOM and XSLT.
 if defined(VBOX_WITH_MAIN) \
