--- configure.in
+++ configure.in
@@ -1360,7 +1360,6 @@ AC_TRY_LINK([#include <stdarg.h>],[va_list valist; vsprintf(0,0,valist);],
 AC_HEADER_DIRENT
 
 AC_MSG_CHECKING(for setenv)
-AH_TEMPLATE([USESETENV], [If your system has setenv() and unsetenv() define USESETENV])
 if test -z "$ac_setenv_args"; then
     AC_TRY_LINK(
     [#include <stdlib.h>],
@@ -1385,9 +1384,6 @@ if test -n "$ac_setenv_args"; then
 else
     AC_MSG_RESULT(no)
     AC_MSG_CHECKING(for putenv)
-    AH_TEMPLATE([NEEDPUTENV],
-    [If your system does not come with a setenv()/putenv()/getenv()
-     functions, you may bring in our own code by defining NEEDPUTENV.])
     AC_TRY_LINK(,[putenv((char *)0);unsetenv((char *)0);], AC_MSG_RESULT(yes) ,  AC_MSG_RESULT(no);AC_DEFINE(NEEDPUTENV))
 fi
 AC_MSG_CHECKING([for nl_langinfo(CODESET)])
